ntnghia/MISA.AMIS.Web/js/common/base.js

```diff
@@ -253,124 +171,188 @@
+                url: me.host + "/api/customergroups",
+                method: 'GET'
+            }).done(function (res) {
+                if (res) {
+                    $.each(res, function (index, obj) {
+                        var option = `<option value="${obj.CustomerGroupId}">${obj.CustomerGroupName}</option>`;
+                        select.append(option);
+                    })
+                }
+            }).fail(function (res) {
 
-        //Lấy khóa chính của bản ghi
-        var recordId = $(this).data('recordId');
-        me.recordId = recordId;
-        console.log(recordId);
-
-        console.log(me.host + me.apiRouter + `/${recordId}`);
-        //Gọi service lấy thông tin chi tiết qua id
-        $.ajax({
-            url: me.host + me.apiRouter + `/${recordId}`,
-            method: "GET"
-        }).done(function (res) {
-            console.log(res);
-            //Binding lên form chi tiết:
-            var elements = $('.dialog-content input[id], select[id]');
-            $.each(elements, function (index, input) {
-                var attr = $(this).attr('id');
-                var value = res[attr];
-                $(this).val(value);
-                //Check trường hợp input là radio button, lấy ra giới tính
-                //if ($(this).attr('type') == "radio") {
-                //    if (this.checked) {
-                //        customer[attr] = value;
-                //    }
-                //} else {
-                //    customer[attr] = value;
-                //}
             })
-        }).fail(function (res) {
 
-        })
+            //Lấy khóa chính của bản ghi
+            var recordId = $(e.currentTarget).data('recordId');
+
+            me.recordId = recordId;
+
+            //Gọi service lấy thông tin chi tiết qua id
+            $.ajax({
+                url: me.host + me.apiRouter + `/${recordId}`,
+                method: "GET"
+            }).done(function (res) {
+                //Binding lên form chi tiết:
+                var elements = $('.dialog-content input[id], select[id]');
+                $.each(elements, function (index, input) {
+                    var attr = $(this).attr('id');
+                    var value = res[attr];
+                    if (value != null) {
+                        if ($(this).attr('type') == 'date') {
+                            var dateConvert = new Date(value);
+                            var day = dateConvert.getDay();
+                            var month = dateConvert.getMonth();
+                            var year = dateConvert.getFullYear();
+                            if (day < 10) {
+                                day = "0" + day;
+                            }
+                            if (month < 10) {
+                                month = "0" + month;
+                            }
 
-        $('.dialog-modal').css("display", "flex");
+                            $(this).val(year + "-" + month + "-" + day);
```

Comment:
Nên tách ra thành 1 hàm có thể sử dụng chung được. Không nên để thế này sau muốn format date dạng này sẽ phải viết lại.
Chương trình của mình có yêu cầu phải định dạng hiển thị ngày/tháng/năm nên sẽ phải dùng ở rất nhiều chỗ. Mình cần tính toán trước các trường hợp như thế này. (Tương tự với định dạng hiển thị tiền tệ)